File: python/Strategy/Duck/FlyBehavior.py
from abc import ABC, abstractmethod


class FlyBehavior(ABC):
    @abstractmethod
    def fly(self):
        pass


class FlyWithWings(FlyBehavior):
    def fly(self):
        print("I'm flying!")


class FlyNoWay(FlyBehavior):
    def fly(self):
        print("I can't fly!")


class FlyRocketPowered(FlyBehavior):
    def fly(self):
        print("I'm flying with a rocket!")

File: c/Strategy/FlyBehavior.c
#include "FlyBehavior.h"

void FlyBehNew(FlyBehavior *flybeh, flyFun fly_fun)
{
    flybeh->fly = fly_fun;
}

void FlyWithWings(FlyBehavior *flybeh)
{
    printf("I'm flying!\n");
}

void FlyNoWay(FlyBehavior *flybeh)
{
    printf("I can't fly!\n");
}

void FlyRocketPowered(FlyBehavior *flybeh)
{
    printf("I'm flying with a rocket!\n");
}
